apps/tng-dictionary/src/components/TermIndex/TermIndex.tsx

```diff
@@ -97,43 +97,56 @@ export default function DataGridDemo(): JSX.Element {
   />;
 
   // SWITCHCOMPONENT
-  const [toggled, setToggled] = React.useState(false);
 
-  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setToggled(event.target.checked)
-    useEffect
+  const [toggled, setToggled] = React.useState(false); //handles switching from const searchTsilhqotin to const searchEnglish
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setToggled(event.target.checked);
+  }
+
+  const handleChange2 = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setComponentState({
+      ...componentState,
+      searchContext: mapSwitchStateToSearchContext(event.target.checked)
+    })
+  }
```

Comment:
Justin- Just reread and study the way state is used \ updated in this file. We made some major changes together in person.